### ISSUE TITLE
Problem found in yosys UHDM Plugin with "vpiParameterTypes"

### ISSUE THREAD

**Issue by QuantamHD** [chipsalliance/Surelog#2560]
```systemverilog
package new_packge;
typedef struct packed {
    logic [128:0] a;
    logic [4:0]   b;
    logic         c;
    logic         d;
    logic         e;
} zzz;
endpackage : new_package;


module module_a
  #(
    parameter type TYPE_PARAMETER = new_package::zzz,
    )
  (
   input TYPE_PARAMETER input_struct,
   output TYPE_PARAMETER output_struct,
   input clk,

   );

  always_ff @(posedge clk) begin
    output_struct <= input_struct;
  end


endmodule : module_a
```

I started off fixing this bug by adding the required vpiTypeParameter handler in the process_object function, but it runs into an issue with the AST in Yosys.

```
    case vpiTypeParameter:
        process_parameter();
        break;
```

Yosys by default expects it's parameters to have a defined size. It fails in UhdmAst.cc:278. The wire_node->simplify call into Yosys tries to run detectSignWidthWorker(width_hint, sign_hint, found_real); on an AST_WIRETYPE and fails.

```
    while (wire_node->simplify(true, false, false, 1, -1, false, false)) {
    }
```

This might be better to handle by turning TypeParameters into typedefs.

https://github.com/SymbiFlow/yosys-symbiflow-plugins/issues/217

**Comment by alaindargelas** [comment]
First let's fix the typos in the test:
```
package new_packge;    <---------------------------------------------------- Wrong package name 
typedef struct packed {
    logic [128:0] a;
    logic [4:0]   b;
    logic         c;
    logic         d;
    logic         e;
} zzz;
endpackage : new_package;


module module_a
  #(
    parameter type TYPE_PARAMETER = new_package::zzz,  <---------------------------------------------------- typo ,
    )
  (
   input TYPE_PARAMETER input_struct,
   output TYPE_PARAMETER output_struct,
   input clk,    <---------------------------------------------------- typo ,

   );

  always_ff @(posedge clk) begin
    output_struct <= input_struct;
  end


endmodule : module_a
```

**Comment by alaindargelas** [comment]
Fixed by #2564
